src/Microsoft.AspNet.Mvc.Core/ActionContext.cs

```diff
@@ -13,6 +14,21 @@ namespace Microsoft.AspNet.Mvc
     public class ActionContext
     {
         /// <summary>
+        /// Creates a empty <see cref="ActionContext"/>.
+        /// </summary>
+        /// <remarks>
+        /// The default constructor should be used only when <see cref="ActionContext"/> needs to be directly
+        /// instantiated in user codes, such as for unit tests.
+        /// </remarks>
+        public ActionContext()
+        {
+            ActionDescriptor = new ActionDescriptor();
+            ModelState = new ModelStateDictionary();
+            HttpContext = new DefaultHttpContext();
+            RouteData = new RouteData();
```

Comment:
The reason why we want to initialize `ModelState` by default is so when a user does
```
if (ModelState.IsValid) { ... }
```

Then the `ActionContext.ModelState` and `ViewContext.ModelState` will be the **same** object. Please write a test for this as well.